x/uibc/quota/keeper/quota_test.go

```diff
@@ -25,3 +41,117 @@ func TestResetQuota(t *testing.T) {
 	assert.NilError(t, err)
 	assert.DeepEqual(t, q.Amount, sdk.NewDec(0))
 }
+
+func TestKeeper_CheckAndUpdateQuota(t *testing.T) {
+	invalidToken := sdk.NewCoin("u/u/umee", sdkmath.NewInt(100))
+	umeeUToken := sdk.NewCoin("u/umee", sdkmath.NewInt(100))
+	atomToken := sdk.NewCoin("atom", sdkmath.NewInt(1000))
+	daiToken := sdk.NewCoin("dai", sdkmath.NewInt(50))
+	// gomock initializations
+	leverageCtrl := gomock.NewController(t)
+	defer leverageCtrl.Finish()
+	leverageMlk := mocks.NewMockLeverageKeeper(leverageCtrl)
+
+	oracleCtrl := gomock.NewController(t)
+	defer oracleCtrl.Finish()
+	oracleMlk := mocks.NewMockOracle(oracleCtrl)
+
+	interfaceRegistry := ctypes.NewInterfaceRegistry()
+	marshaller := codec.NewProtoCodec(interfaceRegistry)
+	ctx, k := initFullKeeper(t, marshaller, nil, leverageMlk, oracleMlk)
+	err := k.ResetAllQuotas(ctx)
+	assert.NilError(t, err)
+
+	// invalid token
+	leverageMlk.EXPECT().ExchangeUToken(ctx, invalidToken).Return(sdk.Coin{}, ltypes.ErrNotUToken).AnyTimes()
+
+	err = k.CheckAndUpdateQuota(ctx, invalidToken.Denom, invalidToken.Amount)
+	assert.ErrorIs(t, err, ltypes.ErrNotUToken)
+
+	// uumee
+	leverageMlk.EXPECT().ExchangeUToken(ctx, umeeUToken).Return(
+		sdk.NewCoin("umee", sdkmath.NewInt(100)),
+		nil,
+	).AnyTimes()
+	leverageMlk.EXPECT().GetTokenSettings(ctx, "umee").Return(ltypes.Token{}, ltypes.ErrNotRegisteredToken).AnyTimes()
+
+	err = k.CheckAndUpdateQuota(ctx, umeeUToken.Denom, umeeUToken.Amount)
+	// returns nil when the error is ErrNotRegisteredToken
+	assert.NilError(t, err)
+
+	// atom
+	leverageMlk.EXPECT().GetTokenSettings(ctx, "atom").Return(
+		lfixtures.Token("atom", "ATOM", 6), nil,
+	).AnyTimes()
+	oracleMlk.EXPECT().Price(ctx, "ATOM").Return(sdk.Dec{}, types.ErrMalformedLatestAvgPrice)
+
+	err = k.CheckAndUpdateQuota(ctx, atomToken.Denom, atomToken.Amount)
+	assert.ErrorIs(t, err, types.ErrMalformedLatestAvgPrice)
+
+	// dai
+	leverageMlk.EXPECT().GetTokenSettings(ctx, "dai").Return(
+		lfixtures.Token("dai", "DAI", 6), nil,
+	).AnyTimes()
+	oracleMlk.EXPECT().Price(ctx, "DAI").Return(sdk.MustNewDecFromStr("0.37"), nil)
+
+	err = k.SetParams(ctx, uibc.DefaultParams())
+	assert.NilError(t, err)
```

Comment:
```suggestion
	err = k.SetParams(ctx, uibc.DefaultParams())
	assert.NilError(t, err)
```

Were params changed before they were set to DefaultParams in this scenario? A comment would clarify whether this is performing a param change required for the test scenario or just testing `SetParams`.